### ISSUE TITLE
Systemd-Notify integration

### ISSUE THREAD

**Issue by TriMoon** [lutris/lutris#5800]
Guys anyone interested in making Lutris more `systemd` friendly?
- On discord unfortunatly @strycore already expressed [not to be interested](https://discord.com/channels/512538904872747018/799769240471207996/1317255188960116817).
  > because it's bloat and out of scope

I have a module that can send notify signals to systemd, and it workds great but im not very familiar with how Lutris works so i don't exactly know where to place the calls...
It's nice to be able to start Lutris as a user service in `notify` mode...
My use case is:
- One main Lutris service, and then sub-units to start `Epic Games Store`, `Wargaming Game Center` and some others at login.

It doesn't make lutris __REQUIRE__ `systemd` but will make use of it when present...

---
For those that are interested in helping, i have created my repo with changes i made so far:
https://gitlab.com/TriMoon/lutris-sd_notify/-/merge_requests/1

---
In current state the main service seems to properly work...

But when a sub-unit is started that starts a game directly using `lutris:rungame/xxx` / `lutris:rungameid/###`;
- That game starts but the unit fails with "protocol error". 🤔
- The unit also only shows "Initialized" as unit status, which indicates `do_command_line()` is not called. 🤔 

No idea where to put the `sd_notify.ready()` call in these cases 🤷‍♀️

**Comment by strycore** [comment]
Use Dbus for your service, we will never accept a systemd based solution because of all the little bitches that have a severe trauma caused by whatever systemd did to them (nothing)

**Comment by TriMoon** [comment]
@strycore 
> we will never accept a systemd based solution

I think i have to point out in bold that this does NOT REQUIRE/DEMAND systemd, it just makes use of it's possibilities when it is found to be available...

That said, where can i find info about which DBus functionality lutris provides?
Maybe i can recode to make the sub-units wait on full startup of the main process using DBus? :thinking:

**Comment by strycore** [comment]
There is no DBus functionality in Lutris because we're not using a client server architecture, this has yet to be done and will likely involve rewriting big chunks of the application and isn't planned to happen any time soon. 

I still have the hardest time trying to wrap my head around what it is you are trying to achieve with this.

**Comment by TriMoon** [comment]
> I still have the hardest time trying to wrap my head around what it is you are trying to achieve with this.

1. Start discod minimized to tray.
2. Start lutris minimized to tray after (1), so discord online presence will work.
3. Start several Game centers minimized to tray after (2), using lutris, like Epic Game Store / Steam / Wargaming Game Center / Ubisoft Connect / Possibly also start some games, each on their designated separate desktop screen.
5. All of this using system-wide user-units on login to desktop instead of desktop-files, and having one place to follow the logs.

**Comment by strycore** [comment]
I still don't understand what is your goal but it doesn't matter because I have a solution for you.

This should be enough.
https://github.com/flattool/ignition


Glad we got this situation sorted in such an elegant way.

**Comment by TriMoon** [comment]
That is completely unusable,because XDG startup files do not provide a way to depend one on another, to start applications after each other.
XDG desktop files are spawning apps in parallel. (at same time)

**Comment by strycore** [comment]
> after (1), so discord online presence will work.

This sounds like something that should get fixed rather than something to build features around

For the rest, I don't get why you need to specify the order of running things. Just open your game launchers on session startup. The order doesn't matter.

> But np, because you are as the maintainer alergic to systemd i won't be spending any more time on this...

I have no problem with systemd, I have problems with useless features that make my software more complex. I'm pretty sure you can do your systemd stuff without touching a single line of the Lutris codebase, and you are encouraged to do so.
